frontend/pages/hosts/ManageHostsPage/ManageHostsPage.tsx

```diff
@@ -688,12 +710,16 @@ const ManageHostsPage = ({
         setSearchQuery(searchText);
       }
 
+      if (!isEqual(page, pageIndex)) {
+        setPage(pageIndex as number);
+      }
+
       // Rebuild queryParams to dispatch new browser location to react-router
       const newQueryParams: { [key: string]: string | number | undefined } = {};
       if (!isEmpty(searchText)) {
         newQueryParams.query = searchText;
       }
-
+      newQueryParams.page = pageIndex as number;
```

Comment:
Can we use the `parseInt` pattern here instead of relying on `as` type coercion?